### PR TITLE
Refine logging in pkg/dashboard/collector/server.go

### LINKED ISSUE #2879: Refine logging in `pkg/dashboard/collector/server.go`, acquired a logger from constructor `NewServer` and use it as the logger for spawned colletors
> Please leave comments with your idea if you are interested in this issue. And comment `/assign` to operate self-assign before you work on it.

### PR DESCRIPTION
**Pull request by z1cheng**
Signed-off-by: z1cheng <imchench@gmail.com>

### What problem does this PR solve?

Close #2879

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: pkg/dashboard/collector/server.go [hunk 1]
```diff
@@ -78,61 +80,61 @@ func NewServer(
...
+			logger.Error(err, "fail to create client pool")
 			os.Exit(1)
 		}
 	}
 
 	for kind, chaosKind := range v1alpha1.AllKinds() {
 		if err = (&ChaosCollector{
 			Client:  s.Manager.GetClient(),
-			Log:     ctrl.Log.WithName("collector").WithName(kind),
+			Log:     logger,
```
Comment (iguoyr) [inline pkg/dashboard/collector/server.go:105]:
```suggestion
			Log:     logger.WithName(kind),
```

**Review by iguoyr — COMMENTED** [review]
Rest LGTM

**Comment by z1cheng** [comment]
> Rest LGTM

@iguoyr Fixed, thanks for reminding~

**Review by iguoyr — APPROVED** [review]
LGTM

**Review by fewdan — APPROVED** [review]
LGTM

**Pull request merged**